core/vm/interface.go

```diff
@@ -24,18 +26,29 @@ import (
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
+type AccountExtraDataStateReader interface {
+	// Return nil for public contract
+	ReadPrivacyMetadata(addr common.Address) (*state.PrivacyMetadata, error)
+	ReadManagedParties(addr common.Address) ([]string, error)
+}
+
+type AccountExtraDataStateWriter interface {
```

Comment:
Question: why do we need to split the interface between reader and writer? Thanks

Reply:
Having them separate and explicit makes it easier to reason for `AccountExtraDataStateReaderByNumber` which returns only the reader for a block height.